src/models/StyleTags.js

```diff
@@ -0,0 +1,396 @@
+// @flow
+/* eslint-disable flowtype/object-type-delimiter */
+/* eslint-disable react/prop-types */
+
+import React from 'react'
+import { IS_BROWSER, DISABLE_SPEEDY, SC_ATTR } from '../constants'
+import { type ExtractedComp } from '../utils/extractCompsFromCSS'
+import { splitByRules } from '../utils/stringifyRules'
+import getNonce from '../utils/nonce'
+
+export interface Tag<T> {
+  // $FlowFixMe: Doesn't seem to accept any combination w/ HTMLStyleElement for some reason
+  styleTag: HTMLStyleElement | null;
+  names: string[];
+  getIds(): string[];
+  insertMarker(id: string): T;
+  insertRules(id: string, cssRules: string[]): void;
+  css(): string;
+  toHTML(additionalAttrs: ?string): string;
+  toElement(): React.Element<*>;
+  clone(): Tag<T>;
+}
+
+/* this error is used for makeStyleTag */
+const parentNodeUnmountedErr =
+  process.env.NODE_ENV !== 'production'
+    ? `
+Trying to insert a new style tag, but the given Node is unmounted!
+- Are you using a custom target that isn't mounted?
+- Does your document not have a valid head element?
+- Have you accidentally removed a style tag manually?
+`.trim()
+    : ''
+
+/* this error is used for tags */
+const throwCloneTagErr = () => {
```

Comment:
Why is this a function, but all the other errors are static strings? This doesn't really need to be a function right?

Reply:
That’s for convenience; look at where it’s used in the style tag creation. I’m avoiding writing some anonymous functions there repeatedly